R/util.R

```diff
@@ -610,3 +610,46 @@ abbreviate <- function(x, len = round(getOption("width", 80) * 0.8)) {
   }
   x
 }
+
+handle_missing_packages <- function(missing_packages) {
+  ## collapse vector to packages to string "c('pckg_1','pckg_2')"
+  vector_packages <- sprintf("install.packages(c(%s))",
+                             paste(sprintf("'%s'", missing_packages),
+                                   collapse=","))
+
+  ## check if we are interactive and logging is active...
+  logging <- !isTRUE(getOption("orderly.nolog"))
+  if (interactive() && logging) {
+    ## ...if so ask if Orderly should try to install the pacakges
```

Comment:
https://codecov.io/gh/vimc/orderly/src/cda69847a1a5ae21365d5b7761adc49ae3900712/R/util.R

Not sure why codecov's bot has not commented on the PR but this section here is untested.  This is the sort of thing that is a nightmare to test though...

Reply:
please pull all the TRUE branch of this clause into its own function (`install_missing_packages` perhaps). Then we can use the mocking package `mockery` to start testing branches in it more easily